### PR TITLE
feat: core changes to support create scratch and sandboxes

### PR DESCRIPTION
**Pull request by peternhale**
@W-10749253@
@W-10657688@
@W-10657657@

### CODE REVIEW

**Review by mshanemc — APPROVED** [review]

File: src/config/configAggregator.ts [hunk 1]
```diff
@@ -170,11 +170,16 @@ export class ConfigAggregator extends AsyncOptionalCreatable<JsonMap> {
    * @param key The key of the property.
    */
   public getPropertyValue<T extends AnyJson>(key: string): Optional<T> {
-    if (this.getAllowedProperties().some((element) => key === element.key)) {
-      return this.getConfig()[key] as T;
-    } else {
+    if (!this.getAllowedProperties().some((element) => key === element.key)) {
```
Comment (mdonnalley) [inline src/config/configAggregator.ts:173]: All these changes to ConfigAggregator shouldn't be needed once https://github.com/salesforcecli/plugin-config/pull/299 is merged

**Review by mdonnalley — CHANGES_REQUESTED** [review]

File: src/config/configAggregator.ts (hunk 1 above)
Comment (peternhale) [reply src/config/configAggregator.ts:173]: How does a change to plugin-config mitigate this change?

Comment (mdonnalley) [reply src/config/configAggregator.ts:173]: Sorry, wrong PR. This is the sfdx-core one: https://github.com/forcedotcom/sfdx-core/pull/563

Comment (peternhale) [reply src/config/configAggregator.ts:173]: @mdonnalley Did you want us to wait for PR #563 to be merged before we merge this one?

Comment (mdonnalley) [reply src/config/configAggregator.ts:173]: @peternhale Yes. I don't want to merge in any of the changes to ConfigAggregator that were made in this PR

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**